app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```

Comment:
```suggestion
  devise :database_authenticatable, :registerable, :validatable
```
E tìm hiểu từng module của devise hỗ trợ những hàm nào luôn nha. Tương tự cho router